xwiki-platform-core/xwiki-platform-xclass/xwiki-platform-xclass-ui/src/main/resources/XWiki/ClassSheet.xml

```diff
@@ -318,6 +323,46 @@
     [[$services.localization.render('platform.xclass.defaultClassSheet.template.view',
       [$templatePath.trim()]) »&gt;&gt;${classTemplateDoc.fullName}]]
   #end
+  ## Create a template provider only if a template for the current class exists.
+  #if ($classTemplateDoc.getObject(${doc.fullName}))
+    (% id="HClassTemplateProvider" %)
+    = {{translation key="platform.xclass.defaultClassSheet.templateProvider.heading"/}} =
+
+      {{info}}$services.localization.render('platform.xclass.defaultClassSheet.templateProvider.description',
+        ['//']){{/info}}
+
+    #if (!$hasClassTemplateProvider)
+      #set ($templateProviderClassName = 'XWiki.TemplateProviderClass')
+      ## Do the page creation and object addition in one step, providing some default values.
+      #set ($restrictionSpace = $doc.documentReference.spaceReferences.get(0).name)
```

Comment:
This looks like a code smell. Will it work with nested pages?

Reply:
At the very least there needs a comment explaining why we're doing this and picking the first space.

Reply:
I added a comment to explain this in https://github.com/xwiki/xwiki-platform/commit/3de26e46b78a651ba27331528dd0c6e9c28eb53d.